app/celery/nightly_tasks.py

```diff
@@ -50,13 +53,13 @@ def remove_letter_csv_files():
 
 def _remove_csv_files(job_types):
     while True:
-        jobs = dao_get_jobs_older_than_data_retention(notification_types=job_types, limit=20000)
+        jobs = dao_get_jobs_older_than_data_retention(notification_types=job_types, limit=100)
         if len(jobs) == 0:
             break
         current_app.logger.info("Archiving {} jobs.".format(len(jobs)))
+        s3.remove_job_batch_from_s3(jobs)
+        dao_archive_job_batch(jobs)
         for job in jobs:
-            s3.remove_job_from_s3(job.service_id, job.id)
-            dao_archive_job(job)
             current_app.logger.info("Job ID {} has been removed from s3.".format(job.id))
```

Comment:
We could dump the list instead of iterating through each individual entry as well.